web/controllers/api/docs_controller.ex

```diff
@@ -163,16 +163,13 @@ defmodule HexWeb.API.DocsController do
     HexWeb.CDN.purge_key(:fastly_hexdocs, unversioned_key)
     HexWeb.CDN.purge_key(:fastly_hexdocs, versioned_key)
 
-    # Set docs flag on release
-    HexWeb.Repo.transaction(fn ->
-      Ecto.Changeset.change(release, has_docs: true)
-      |> HexWeb.Repo.update!
+    multi =
+      Ecto.Multi.new
+      |> Ecto.Multi.update(:release, Ecto.Changeset.change(release, has_docs: true))
+      |> Ecto.Multi.update(:package, Ecto.Changeset.change(release.package, docs_updated_at: Ecto.DateTime.utc))
+      |> Ecto.Multi.run(:log, fn _ -> audit(user, "docs.publish", {package, release}) end)
```

Comment:
Yes, you are right. put_assoc does not call changeset, sorry about that. My idea about using `logger_id` was because Ecto would complain if such field was not defined. BUT if Ecto is not complaining, than you don't need it.

> Btw, would you allow setting just nil and [] or actual data as well, as in arbitrary records or arrays of records?

I guess we should support whatever if we decide to add this feature. Would you please send a PR?